pkg/services/ngalert/schedule/schedule.go

```diff
@@ -443,9 +444,15 @@ func (sch *schedule) schedulePeriodic(ctx context.Context) error {
 				item := readyToRun[i]
 
 				time.AfterFunc(time.Duration(int64(i)*step), func() {
-					success := item.ruleInfo.eval(tick, item.version)
+					success, dropped := item.ruleInfo.eval(tick, item.version)
 					if !success {
 						sch.log.Debug("scheduled evaluation was canceled because evaluation routine was stopped", "uid", item.key.UID, "org", item.key.OrgID, "time", tick)
+						return
+					}
+					if dropped != nil {
+						sch.log.Warn("Alert rule evaluation is too slow - dropped tick", "uid", item.key.UID, "org", item.key.OrgID, "time", tick)
+						orgID := fmt.Sprint(item.key.OrgID)
+						sch.metrics.EvaluationMissed.WithLabelValues(orgID, item.key.UID, item.ruleName).Inc()
```

Comment:
The other small thing Is that you can probably refactor:

```golang
sch.metrics.EvaluationMissed.WithLabelValues(orgID, item.key.UID, item.ruleName).Inc()

```
to be a method on `readyToRunItem`

I can't remember if `WithLabelValues` is optimised for lesser allocations on equal values (it probably is) so we don't gain any performance improvement. but I think it looks nice that you could end up doing:
`
``
item.MissedEvaluationInc()
```

or even `item.MissedEvaluation.Inc()` if you simply want to expose the metric directly from `readyToRunItem`